File: ts/project/.sdk/tm/php/core/Operation.php
<?php
declare(strict_types=1);

// ProjectName SDK operation

class ProjectNameOperation
{
    public string $entity;
    public string $name;
    public string $input;
    public array $points;
    public ?array $alias_map;

    public function __construct(array $opmap = [])
    {
        $e = \Voxgig\Struct\Struct::getprop($opmap, 'entity');
        $this->entity = (is_string($e) && $e !== '') ? $e : '_';
        $n = \Voxgig\Struct\Struct::getprop($opmap, 'name');
        $this->name = (is_string($n) && $n !== '') ? $n : '_';
        $i = \Voxgig\Struct\Struct::getprop($opmap, 'input');
        $this->input = (is_string($i) && $i !== '') ? $i : '_';

        $this->points = [];
        $raw_points = \Voxgig\Struct\Struct::getprop($opmap, 'points');
        if (is_array($raw_points)) {
            foreach ($raw_points as $t) {
                if (is_array($t)) {
                    $this->points[] = $t;
                }
            }
        }

        $raw_alias = \Voxgig\Struct\Struct::getprop($opmap, 'alias');
        $this->alias_map = is_array($raw_alias) ? $raw_alias : null;
    }
}

File: ts/project/.sdk/tm/csharp/core/Operation.cs
// ProjectName SDK - resolved operation definition.

using Voxgig.Struct;

namespace ProjectNameSdk;

public class Operation
{
    public string Entity = "_";
    public string Name = "_";
    public string Input = "_";
    public List<Dictionary<string, object?>> Points = new();
    public Dictionary<string, object?>? Alias;

    public Operation(Dictionary<string, object?> opmap)
    {
        if (StructUtils.GetProp(opmap, "entity") is string entity && entity != "")
        {
            Entity = entity;
        }
        if (StructUtils.GetProp(opmap, "name") is string name && name != "")
        {
            Name = name;
        }
        if (StructUtils.GetProp(opmap, "input") is string input && input != "")
        {
            Input = input;
        }

        var rawPoints = StructUtils.GetProp(opmap, "points");
        if (rawPoints is List<object?> tlist)
        {
            foreach (var t in tlist)
            {
                if (t is Dictionary<string, object?> tm)
                {
                    Points.Add(tm);
                }
            }
        }
        else if (rawPoints is List<Dictionary<string, object?>> mlist)
        {
            Points = mlist;
        }

        if (StructUtils.GetProp(opmap, "alias") is Dictionary<string, object?> am)
        {
            Alias = am;
        }
    }
}
